rs/canonical_state/tests/compatibility.rs

```diff
@@ -71,33 +68,10 @@ pub(crate) fn arb_valid_versioned_stream_header(
     max_signal_count: usize,
 ) -> impl Strategy<Value = (StreamHeader, RangeInclusive<CertificationVersion>)> {
     prop_oneof![
-        // Stream headers up to version 18 may have reject signals for responses
-        // (`CanisterMigrating` only) and the `DeprecatedResponsesOnly` flag set.
-        (
-            arb_stream_header(
-                /* min_signal_count */ 0,
-                max_signal_count,
-                /* with_reject_reasons */ vec![RejectReason::CanisterMigrating],
-            ),
-            Just(MIN_SUPPORTED_CERTIFICATION_VERSION..=CertificationVersion::V18)
-        ),
         // Stream headers may have flavours of reject signals other than `CanisterMigrating`
         // starting from certification version 19.
         (
-            arb_stream_header(
-                /* min_signal_count */ 0,
-                max_signal_count,
-                /* with_reject_reasons */
-                vec![
-                    RejectReason::CanisterMigrating,
-                    RejectReason::CanisterNotFound,
-                    RejectReason::CanisterStopped,
-                    RejectReason::CanisterStopping,
-                    RejectReason::QueueFull,
-                    RejectReason::OutOfMemory,
-                    RejectReason::Unknown
-                ],
-            ),
```

Comment:
I would retain the `with_reject_reasons` argument to `arb_stream_header()`, as we may be introducing new signals in the future. I'd rather have some redundancy in the code now (we could use `RejectReason::all()`, so it's less verbose) than potentially end up with a much more cumbersome approach if/when we do need to pick and choose signal types later one.